### PR TITLE
Fix default behaviour of enter press leading to form submission in core components.

### PR DESCRIPTION
**Pull request by muskgupta**
## Description
Pressing enter on core components redirects to thankyou page which is not the correct behaviour. This fix prevents the default behaviour of enter press on form level.

## Related Issue
https://jira.corp.adobe.com/browse/FORMS-15630

## How Has This Been Tested?
Tested on localhost deployment and test cases tested using cypress.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

**Review by rismehta — CHANGES_REQUESTED** [review]
Can we also change the type of submit button in this PR, since the changes are related.

**Comment by muskgupta** [comment]
> Can we also change the type of submit button in this PR, since the changes are related.

As discussed in scrum, will cover in separate PR.

File: ui.frontend/src/view/FormContainer.js [hunk 1]
```diff
@@ -35,6 +35,23 @@ class FormContainer {
...
+
+        // Prevent default behaviour on form container.
+        this.preventDefaultSubmit();
+    }
+
+    /**
+     * Prevents the default behavior of the Enter key on components within the formContainer
+     * from triggering a form submission and redirecting to the Thank-You Page.
+     */
+    preventDefaultSubmit(){
```
Comment (rismehta) [inline ui.frontend/src/view/FormContainer.js:47]: This should be a private API starting with #

File: ui.frontend/src/view/FormContainer.js [hunk 2]
```diff
@@ -35,6 +35,23 @@ class FormContainer {
...
+        if(this._element) {
+            this._element.addEventListener('keydown', function (event) {
```
Comment (rismehta) [inline ui.frontend/src/view/FormContainer.js:49]: Can you arrow function for the event listener, it provides a cleaner syntax

**Review by rismehta — CHANGES_REQUESTED** [review]
check comments

File: ui.frontend/src/view/FormContainer.js (hunk 1 above)
Comment (muskgupta) [reply ui.frontend/src/view/FormContainer.js]: Done.

File: ui.frontend/src/view/FormContainer.js (hunk 2 above)
Comment (muskgupta) [reply ui.frontend/src/view/FormContainer.js]: Sure, updated.

File: ui.tests/test-module/specs/formcontainer.spec.js [hunk 3]
```diff
@@ -287,4 +287,27 @@ describe('Page/Form Authoring', function () {
...
+            const pagePath = "/content/forms/af/core-components-it/samples/emailinput/basic.html";
+
+            beforeEach(function () {
+                cy.previewForm(pagePath);
+            });
+
+            it('check the preventDefaultSubmit method by simulating keydown event on the form', function () {
+                cy.get('.cmp-adaptiveform-container').then((formContainer) => {
+                    cy.stub(formContainer[0], 'onsubmit').as('submit');
+                    cy.get('form').trigger('keydown', {key: 'Enter'});
```
Comment (sakshi-arora1) [inline ui.tests/test-module/specs/formcontainer.spec.js:301]: trigger"enter" on button and input as well for completeness of testing

Comment (muskgupta) [reply ui.tests/test-module/specs/formcontainer.spec.js:301]: Added these in the check below.

**Review by sakshi-arora1 — APPROVED** [review]

**Comment by muskgupta** [comment]
Closed this PR and opened new https://github.com/adobe/aem-core-forms-components/pull/1363 as per discussion in scrum.

**Pull request closed without merging**